Main.lean

```diff
@@ -1,24 +1,35 @@
 import KLR
 import Cli
-import KLR.Python
-import KLR.Trace
 open Cli
 
-
 local instance : MonadLift Err IO where
   monadLift
     | .ok x => return x
     | .error s => throw $ .userError s
 
-def parseJson (p : Parsed) : IO UInt32 := do
+private def parse (p : Parsed) : IO KLR.Python.Kernel := do
   let file := p.positionalArg! "file" |>.as! String
   let s <- IO.FS.readFile file
-  let kernel <- KLR.Python.Parsing.parse s
-  let stmts <- KLR.Trace.runNKIKernel kernel.inferArguments
-  let json := Lean.Json.arr (stmts.map Lean.toJson).toArray
+  KLR.Python.Parsing.parse s
+
+def parseJson (p : Parsed) : IO UInt32 := do
+  let kernel <- parse p
+  let klr <- KLR.Trace.runNKIKernel kernel.inferArguments
+  let json := Lean.toJson klr
   IO.println json
   return 0
 
+def trace (p : Parsed) : IO UInt32 := do
+  let kernel <- parse p
+  let klr <- KLR.Trace.runNKIKernel kernel.inferArguments
+  if p.hasFlag "repr" then
+    IO.println (toString $ repr klr)
+  else if p.hasFlag "json" then
+    IO.println (toString $ Lean.toJson klr)
+  else
+    IO.println (Lean.format klr).pretty
+  return 0
+
 def parseJsonCmd := `[Cli|
```

Comment:
Let's mark this as deprecated, since it seems subsumed by `trace`. Will need to push the change through the interop code but that should be a one liner.